### PR TITLE
Document table row classes & provenance classes

### PR DESCRIPTION
**Pull request by hyanwong**
A proposal for fixing #1921. Not point in repeating documentation here, I reckon, so I'm just pointing to the equivalent attributes in the tree sequence container class.

### CODE REVIEW

File: docs/python-api.md [hunk 1]
```diff
@@ -1243,6 +1243,18 @@ Also see the {ref}`sec_tables_api_table_collection` summary.
     :special-members: __getitem__
 ```
 
+##### Associated row class
+
+A row returned from an {class}`IndividualTable` is an instance of the following
+simple {mod}`dataclass<python:dataclasses>`. Each attribute in this class matches
```
Comment (jeromekelleher) [inline docs/python-api.md:1249]: I don't think we want to document them as dataclasses - we might want to change this implementation detail in the future.

File: python/tskit/tables.py [hunk 2]
```diff
@@ -64,11 +64,15 @@ class NOTSET(metaclass=NotSetMeta):
...
+    """
+    A row in an :class:`IndividualTable`.
+    """
+
     __slots__ = ["flags", "location", "parents", "metadata"]
-    flags: int
-    location: np.ndarray
-    parents: np.ndarray
-    metadata: Optional[Union[bytes, dict]]
+    flags: int  #: See :attr:`Individual.flags`
```
Comment (jeromekelleher) [inline python/tskit/tables.py:72]: Agree there's no point in duplicating here.

Might was well use the standard docstring form though, like
```
""" See :attr:`Individual.flags` """
flags: int
```
as there will be cases in which the form you have will be too long for a single line, and it's good to keep things uniform, even if just for greppability.

**Review by jeromekelleher — COMMENTED** [review]
LGTM, minor comments

File: docs/python-api.md (hunk 1 above)
Comment (hyanwong) [reply docs/python-api.md:1249]: Yeah, that's a good point.

**Comment by hyanwong** [comment]
Now fixes #1921. Note that this PR also adds documentation to the Provenance class.

File: python/tskit/trees.py [hunk 3]
```diff
@@ -603,10 +603,22 @@ def __eq__(self, other):
 
 @dataclass
 class Provenance(util.Dataclass):
+    """
+    A provenance entry in a tree sequence, detailing what how this tree
+    sequence was generated, or what has been done to it (see :ref:`sec_provenance`).
+    """
+
```
Comment (benjeffery) [inline python/tskit/trees.py:610]:
```suggestion
    """
    A provenance entry in a tree sequence, detailing how this tree
    sequence was generated, or subsequent operations on it (see :ref:`sec_provenance`).
    """

```

**Review by benjeffery — APPROVED** [review]
LGTM, one sentence didn't make sense.

File: python/tskit/trees.py (hunk 3 above)
Comment (hyanwong) [reply python/tskit/trees.py:610]: Oops. Editing error. Nice spot, thanks.

**Review by jeromekelleher — APPROVED** [review]

**Comment by hyanwong** [comment]
I forgot to take this off draft mode. Done now, so can be flagged for merge.

**Pull request merged**